autonat.go

```diff
@@ -75,9 +77,9 @@ func NewAutoNAT(ctx context.Context, h host.Host, getAddrs GetAddrs) AutoNAT {
 		getAddrs = h.Addrs
 	}
 
-	emitUnknown, _ := h.EventBus().Emitter(new(event.EvtLocalRoutabilityUnknown))
-	emitPublic, _ := h.EventBus().Emitter(new(event.EvtLocalRoutabilityPublic))
-	emitPrivate, _ := h.EventBus().Emitter(new(event.EvtLocalRoutabilityPrivate))
+	emitUnknown, _ := h.EventBus().Emitter(new(event.EvtLocalRoutabilityUnknown), eventbus.Stateful)
+	emitPublic, _ := h.EventBus().Emitter(new(event.EvtLocalRoutabilityPublic), eventbus.Stateful)
+	emitPrivate, _ := h.EventBus().Emitter(new(event.EvtLocalRoutabilityPrivate), eventbus.Stateful)
```

Comment:
How will this interact with a subscriber who subscribes to all three events? Won't they just be confused as to the current state?

Is there a reason we don't have a single emitter with multiple different state options, instead of three emitters?

Reply:
That's a good point.

Reply:
Depends on https://github.com/libp2p/go-libp2p-core/pull/126

Reply:
We've agreed to merge all three types into a single type with an enum and replace all the emitters with a single emitter. @Stebalien  is working on a PR for that and will fix this PR once this is done.